### PR TITLE
Game statistics methods

### PR DESCRIPTION
**Pull request by GregJMitchell**
This Branch:
Adds
-highest_total_score 
-lowest_total_score
-percentage_home_wins
-percentage_visitor_wins
-percentage_ties
-count_of_games_by_season
-average_goals_per_game
-average_goals_by_season

### CODE REVIEW

File: lib/game_statistics.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
+module GameStatistics
+  def highest_total_score
+    highest_score = highest_total_score_helper
+    highest_score.away_goals + highest_score.home_goals
+  end
+
```
Comment (s-steel) [inline lib/game_statistics.rb:6]: Really like how small and concise these methods are.

File: lib/game_statistics.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    seasons_hash = {}
+    group_by_season.each do |season|
+      total = 0
+      season[1].each do |game|
+        total += game.away_goals + game.home_goals
+      end
+      seasons_hash[season[0]] = (total/season[1].length.to_f).round(2)
+    end
+    seasons_hash
+  end
```
Comment (s-steel) [inline lib/game_statistics.rb:94]: We could look into refactoring this method.

File: test/game_statistics_test.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  def test_it_can_find_average_goals_per_game
+    assert_equal 4.28, @stat_tracker.average_goals_per_game
+  end
+
+  def test_it_can_find_average_goals_per_season
+    expected = {'20172018' => 4.41,
+                '20132014' => 3.81,
+                '20122013' => 5.0 }
+    assert_equal expected, @stat_tracker.average_goals_per_season
+  end
```
Comment (s-steel) [inline test/game_statistics_test.rb:57]: Nice tests.  I was wondering is we should add multiple tests for each method.

**Review by s-steel — APPROVED** [review]
Looks good!  I really like how small most of your methods are.  Maybe just some minor refactoring needed.

File: lib/game_statistics.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+  def average_goals_per_game
+    total = 0
+    @games.each do |game|
+      average_goals = (game.away_goals + game.home_goals) / @games.length.to_f
+      total += average_goals
+    end
+    total.round(2)
+  end
+
```
Comment (cpfergus1) [inline lib/game_statistics.rb:77]: Nice method here, I am sure we can refactor some of my code with this later!

File: lib/game_statistics.rb [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    @games.min_by do |game|
+      game.away_goals + game.home_goals
+    end
+  end
+
+  def percentage_home_wins
+    percent_wins = percentage_home_win_helper
+    ((percent_wins / @games.length.to_f) * 100).round(2)
+  end
+
```
Comment (cpfergus1) [inline lib/game_statistics.rb:28]: Very concise, I like it!

**Review by cpfergus1 — APPROVED** [review]

**Pull request merged**